src/NHibernate/Cache/CacheLock.cs

```diff
@@ -85,10 +85,20 @@ public bool IsPuttable(long txTimestamp, object newVersion, IComparer comparator
 			{
 				return false;
 			}
-			return Version == null ?
-			       UnlockTimestamp < txTimestamp :
-			       comparator.Compare(Version, newVersion) < 0;
-				//by requiring <, we rely on lock timeout in the case of an unsuccessful update!
+
+			return Version == null 
+				? !minimalPut && UnlockTimestamp < txTimestamp 
+				: comparator.Compare(Version, newVersion) < (minimalPut ? 0 : 1);
+			//by requiring <, we rely on lock timeout in the case of an unsuccessful update!
```

Comment:
With your change this comment is no more making sense when `!minimalPut`.

```suggestion
			//by requiring < 0, we rely on lock timeout in the case of an unsuccessful update!
```